CMSgov/easi-app
# Enable Rich text (FE) – TRB

# EASI-3120

## Changes and Description

[TRB Rich text fields pointed out in figma](https://www.figma.com/file/iGatgTnlzTuqW9CQFYHDam/TRB-Intake-e2e?type=design&node-id=4880-131849&mode=design&t=jAZAWw798J00ubD9-0)
- Ready for consult action feedback field
- Request edits action “type of edits” field
- Advice letter meeting summary field
- Advice letter recommendation description field
- Advice letter next steps field
- _Do not do per AC_: Admin experience Admin notes field (on the notes sub-page and in side panel)

## Notes
- There is an additional check on input content changes for empty string values `''`
  - Deleting out the rte field after entering some input will leave behind something like `<p><br></p>`
  - A side effect worth mentioning is that multiple line breaks will be considered an empty field
- Unit test methods needed some tweaks
  - Creating accessibility-minded helpers for these can be done later
  - Query by test id instead of role
  - toContainHTML() instead of toHaveValue()
- Test warnings will show up over a missing ref to `window.computedStyle`. Filling this function did not work. Looks like it could be an issue with the sublib jsdom. Can look into it later.

## Testing
- Updated existing unit tests to reflect rte test methods
- Check Request Edits and Ready for Consult actions for the rte field
- Check rte fields throughout the Advice Letter process
